### PR TITLE
Added solutions to all challenge probelms from codechef

### PR DESCRIPTION
**Pull request by NavyaIyanampudi**
Please find all the codes below!
Hope it helps

### CODE REVIEW

File: CodechefChallengeSolutions/APRIL20B/SQRDSUB.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
+#include <iostream>
```
Comment (amanBhawsar) [inline CodechefChallengeSolutions/APRIL20B/SQRDSUB.cpp]: already there

File: CodechefChallengeSolutions/APRIL20B/STRNO.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#include <bits/stdc++.h>
```
Comment (amanBhawsar) [inline CodechefChallengeSolutions/APRIL20B/STRNO.cpp]: already there

File: CodechefChallengeSolutions/APRIL20B/UNITGCD.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#include <bits/stdc++.h>
+using namespace std;
+/* set<int>v;
+int prime(int n)
+{
```
Comment (amanBhawsar) [inline CodechefChallengeSolutions/APRIL20B/UNITGCD.cpp]: remove unwanted code

File: CodechefChallengeSolutions/DEC19B/WATSCORE.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+void init();
+void init()
+{
+    int i;
+	for( i=1;i<=11;i++)
+	{
+	    d[i]=NULL;
+	}
+}
+long long maxi(int);
```
Comment (amanBhawsar) [inline CodechefChallengeSolutions/DEC19B/WATSCORE.cpp:19]: no need of line no 20

File: CodechefChallengeSolutions/DEC19B/WATSCORE.cpp [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+#include<vector>
+#include<algorithm>
+using namespace std;
+typedef struct node
+{
+    int data;
+    struct node *next;
+}node;
+node *d[11];
+void init();
```
Comment (amanBhawsar) [inline CodechefChallengeSolutions/DEC19B/WATSCORE.cpp]: no need of line no 11

File: CodechefChallengeSolutions/JULY20B/CHEFSTR1.cpp [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	    long long c=0;
+	    for(int i=1;i<n;i++)
+	    {
+	        if(a[i]>a[i-1])
+	        c+=a[i]-a[i-1]-1;
+	        else
+	        c+=a[i-1]-a[i]-1;
+	    }
+	    cout<<c;
+	    cout<<"\n";
```
Comment (amanBhawsar) [inline CodechefChallengeSolutions/JULY20B/CHEFSTR1.cpp]: you could have written it in single cout statement

File: CodechefChallengeSolutions/LTIME74B/RPD.c [hunk 7]
```diff
@@ -0,0 +1,30 @@
+#include<stdio.h>
+int main()
+{
+    int a[1000],sum=0,q=0,k,n,i,j,max=0,num=0,digit=0,m=0,temp=0;
```
Comment (amanBhawsar) [inline CodechefChallengeSolutions/LTIME74B/RPD.c]: remove unused variables like temp

File: CodechefChallengeSolutions/LTIME81B/POPGATES.cpp [hunk 8]
```diff
@@ -0,0 +1,36 @@
+#include <bits/stdc++.h>
+# define  ll long long
+# define max 10000000
+using namespace std;
+ll r[max];
+int main() {
+    ll  w, a,b,c,d,n,m,k,t;
```
Comment (amanBhawsar) [inline CodechefChallengeSolutions/LTIME81B/POPGATES.cpp]: remove unused var a,b,c etc

File: CodechefChallengeSolutions/MARCH20B/ENGXOR.cpp [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        n>>=1;
+    }
+    if(c%2==0)
+    return 1;
+    else
+    return 0;
+}
+int main() {
+	ios_base::sync_with_stdio(false);
+	cin.tie(NULL);
```
Comment (amanBhawsar) [inline CodechefChallengeSolutions/MARCH20B/ENGXOR.cpp]: if you are using these 2 lines then no need to use scanf/printf you can directly use cin/cout with same i/o speed

File: CodechefChallengeSolutions/UWCOI20/UWCOI20A.cpp [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+	// your code goes here
+	int t,n;
+	cin>>t;
+	while(t--)
+	{
+	cin>>n;
+	int a[n];
+	for(int i=0;i<n;i++)
+	{
+	    cin>>a[i];
```
Comment (amanBhawsar) [inline CodechefChallengeSolutions/UWCOI20/UWCOI20A.cpp:14]: you could have done it in O(1) space if compared the element while taking input.

**Review by amanBhawsar — CHANGES_REQUESTED** [review]
Remove the code which are already there. 
I have added some comments kindly resolve them and raise PR.
Good Luck

File: CodechefChallengeSolutions/APRIL20B/SQRDSUB.cpp (hunk 1 above)
Comment (NavyaIyanampudi) [reply CodechefChallengeSolutions/APRIL20B/SQRDSUB.cpp]: Deleted!

File: CodechefChallengeSolutions/DEC19B/WATSCORE.cpp (hunk 4 above)
Comment (NavyaIyanampudi) [reply CodechefChallengeSolutions/DEC19B/WATSCORE.cpp:19]: Done

File: CodechefChallengeSolutions/MARCH20B/ENGXOR.cpp (hunk 9 above)
Comment (NavyaIyanampudi) [reply CodechefChallengeSolutions/MARCH20B/ENGXOR.cpp]: Sure

File: CodechefChallengeSolutions/UWCOI20/UWCOI20A.cpp (hunk 10 above)
Comment (NavyaIyanampudi) [reply CodechefChallengeSolutions/UWCOI20/UWCOI20A.cpp:14]: Will *max_element take any space I mean other than constant space?

**Review by NavyaIyanampudi — COMMENTED** [review]
I made all the changes as you said.
Thank you for your time really helped me

**Comment by NavyaIyanampudi** [comment]
@amanBhawsar Please check and confirm

File: CodechefChallengeSolutions/UWCOI20/UWCOI20A.cpp (hunk 10 above)
Comment (amanBhawsar) [reply CodechefChallengeSolutions/UWCOI20/UWCOI20A.cpp:14]: Nope I was referring to the input array

**Review by amanBhawsar — APPROVED** [review]

**Pull request merged**